app/controllers/posts_controller.rb

```diff
@@ -590,6 +591,10 @@ def revert
         0
       ] if post_revision.modifications["category_id"].present? &&
         post_revision.modifications["category_id"][0] != topic.category.id
+      changes[:tags] = post_revision.modifications["tags"][0].split(
+        ",",
+      ) if post_revision.modifications["tags"].present? &&
+        post_revision.modifications["tags"][0] != topic.tags.pluck(:name).sort.join(",")
```

Comment:
is post_revision.modifications["tags"][0] guaranteed to be in sorted order?

It also looks like you're comparing an array `post_revision.modifications["tags"][0]` to a string `topic.tags.pluck(:name).sort.join(",")`

Reply:
That's unnecessary - post_revision modifications come in as an array from the model.
topic.tags.pluck(:name) comes in as an array, through active record.

It's possible tags were handled as a raw comma separated string in an earlier version of Discourse, but it is not necessary to handle that case today.